skafka/src/main/scala/com/evolutiongaming/skafka/consumer/Consumer.scala

```diff
@@ -563,6 +567,14 @@ object Consumer {
           offsetsOf(partitions) { consumer.endOffsets(_, timeoutJ) }
         }
 
+        def currentLag(partition: TopicPartition): F[Option[Long]] = {
+          val lag = consumer.currentLag(partition.asJava)
```

Comment:
and you got a race conditions here - all calls to Java consumer here should be protected, either by `serialNonBlocking` or `serialBlocking`. Judging by the scaladoc - the method is not blocking so it should be under `serialNonBlocking`

Reply:
I was under impression that `currentLag` is just simple state reading from `consumer`...

but I will wrap it